compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantTypeChecker.java

```diff
@@ -1355,8 +1355,9 @@ private BType checkTupleType(BTupleType tupleType, BLangListConstructorExpr list
     }
 
     private BTupleType createNewTupleType(Location pos, List<BType> memberTypes, AnalyzerData data) {
-        BTypeSymbol tupleTypeSymbol = Symbols.createTypeSymbol(SymTag.TUPLE_TYPE, 0, Names.EMPTY,
-                data.env.enclPkg.symbol.pkgID, null, data.env.scope.owner, pos, SOURCE);
+        BTypeSymbol tupleTypeSymbol =
+                Symbols.createTypeSymbol(SymTag.TUPLE_TYPE, data.constantSymbol.flags, Names.EMPTY,
+                        data.env.enclPkg.symbol.pkgID, null, data.env.scope.owner, pos, SOURCE);
```

Comment:
Here, By setting constant symbol flags to `type symbol`, we set the `constant` flag with the same type symbol as well. Is it correct?

Reply:
In offline, we discuss use the `public` flag instead of setting all the flags in the `constantSymbol` to type symbol. Here, we need to verify whether the public flag is present in the `constantSymbol`.

Reply:
In the symbol resolver we have a check to find out whether the constant we are accessing is Public or not. Thus at this stage we might not need the check to see if the public flag is present.